### PR TITLE
Gdr 2248

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Related JIRA issue: GDR-2248

## Why was it changed?
Improve logic for normalization and identification of single-agent and matrix data with Drug3

# Checklist for sustainable code base
- [X] I added tests for any code changed/added
- [X] I added documentation for any code changed/added
- [X] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: R/map_df.R [hunk 1]
```diff
@@ -186,9 +186,12 @@ map_df <- function(trt_md,
 #' 
 #' @return list
 #' 
-.map_references <- function(mat_elem) {
+.map_references <- function(mat_elem, 
+                            rowData_colnames = c(gDRutils::get_env_identifiers('duration'), 
+                                                paste0(c("drug", "drug_name", "drug_moa"), '3'))) {
```
Comment (j-smola) [inline R/map_df.R:191]: Could you add some test for this new param?

File: R/generate_wrappers.R [hunk 2]
```diff
@@ -6,7 +6,7 @@ generateNoNoiseRawData <- function(cell_lines, drugs, save = TRUE) {
   #### - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   # generate the data for the 1st test set: no noise
   #   only for testing purpuses not displayed as example
-  df_merged <- prepareMergedData(cell_lines[2:11, ], drugs[2:11, ], 0)
+  df_merged <- gDRtestData::prepareMergedData(cell_lines[2:11, ], drugs[2:11, ], 0)
```
Comment (j-smola) [inline R/generate_wrappers.R:9]: This and next functions are not exported from `gDRtestData` :(

**Review by j-smola — COMMENTED** [review]
Please, run `gDRstyle::checkPackage("gDRcore", "./", bioc_check = TRUE)`
and update version in DESCRIPTION to gDRtestData (>= 1.1.2)

File: tests/testthat/test-data_type.R [hunk 3]
```diff
@@ -41,3 +41,29 @@ test_that("identify_data_type and split_raw_data works as expected", {
   split_df2 <- split_raw_data(df2)
   expect_true(all(split_df2[["Gnumber"]] == "DrugA"))
 })
+
+
+test_that("collapse drugs works as expected", {
```
Comment (gladkia) [inline tests/testthat/test-data_type.R:46]: 👍

**Review by gladkia — APPROVED** [review]

File: R/data_type.R [hunk 4]
```diff
@@ -271,3 +276,41 @@ unify_combination_data <- function(df, cl, drug_ids) {
...
+#'
+collapse_drugs <- function(df) {
+  
+  drug_ids <- unlist(gDRutils::get_env_identifiers(
+    c("drug_name", "drug_name2", "drug_name3", "drug", "drug2", "drug3",
+      "drug_moa", "drug_moa2", "drug_moa3", "concentration", "concentration2",
+      "concentration3"), 
+    simplify = FALSE)
+  )
+  # add '1' to the names for more efficient loop lower
```
Comment (gladkia) [inline R/data_type.R:294]: 👍

File: tests/testthat/test-data_type.R [hunk 5]
```diff
@@ -41,3 +41,29 @@ test_that("identify_data_type and split_raw_data works as expected", {
...
+  idfs <- get_env_identifiers()
```
[the remark this answers is not in the document]
Comment (j-smola) [reply tests/testthat/test-data_type.R]: ✔️

File: R/generate_wrappers.R (hunk 2 above)
Comment (j-smola) [reply R/generate_wrappers.R:9]: ✔️

File: R/data_type.R [hunk 6]
```diff
@@ -271,3 +276,42 @@ unify_combination_data <- function(df, cl, drug_ids) {
...
+  names(drug_ids)[!grepl('[0-9]', names(drug_ids))] <- paste0(names(drug_ids)[!grepl('[0-9]', names(drug_ids))], '1')
```
[the remark this answers is not in the document]
Comment (j-smola) [reply R/data_type.R]: ✔️

**Review by j-smola — APPROVED** [review]
GJ!

**Pull request merged**